src/main/java/com/github/webicitybrowser/spec/fetch/imp/FetchEngineImp.java

```diff
@@ -41,8 +45,36 @@ private FetchResponse httpNetworkFetch(FetchParams params) {
 
 	private void fetchResponseHandover(FetchParams params, FetchResponse response) {
 		if (params.consumeBodyAction() != null) {
-			// TODO: Queue a response
+			if(response.body() == null) {
+				queueAFetchTask(() -> params.consumeBodyAction().execute(response, true, new byte[] {}), params);
+			} else {
+				fullyReadBody(params, response);
+			}
 		}
 	}
-	
+
+	private void queueAFetchTask(Runnable fetchTask, FetchParams params) {
+		params.taskDestination().enqueue(fetchTask);
+	}
+
+	private void fullyReadBody(FetchParams params, FetchResponse response) {
+		//TODO: implement fully read
+		params.taskDestination().dequeueAll();
+
+		try {
+			int next;
+			BufferedReader br = new BufferedReader(response.body().readableStream());
+			ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+			while((next = br.read()) != -1) {
```

Comment:
The read all bytes algorithm is defined by the stream spec, as opposed to by the fetch spec. You may want to create a new package and utility class for this